.github/workflows/check_plagiarism.yml

```diff
@@ -15,37 +15,72 @@ jobs:
         with:
           fetch-depth: 0
 
-      - name: Set up Python
-        uses: actions/setup-python@v5
+      - name: Set up Node.js
+        uses: actions/setup-node@v4
         with:
-          python-version: '3.10'
+          node-version: '18.20.4'
 
-      - name: Install Compare50 && beautifulsoup4
-        run: pip install compare50 beautifulsoup4
+      - name: Delete existing package.json
+        run: |
+          if [ -f "package.json" ]; then
+            rm package.json
+          fi
+
+      - name: Create new package.json for plagiarism checker
+        run: |
+          cat > package.json <<EOL
+          {
+            "name": "plagiarism-checker",
+            "version": "1.0.0",
+            "description": "A script to check for plagiarism using MOSS.",
+            "main": ".github/scripts/plagiarism_check.cjs",
+            "dependencies": {
+              "axios": "^1.0.0",
+              "cheerio": "^1.0.0"
+            },
+            "scripts": {
+              "start": "node .github/scripts/plagiarism_check.cjs"
+            }
+          }
+          EOL
+
+      - name: Install Dependencies for Plagiarism Checker
+        run: npm install
 
       - name: Get list of changed files
         id: changed-files
+        run: |
+          js_files=$(git diff --name-only --diff-filter=AM --find-renames --find-copies ${{ github.event.pull_request.base.sha }}..${{ github.event.pull_request.head.sha }} | grep '^games/.*\.js$' | xargs)
+          echo "FILES=$js_files" >> $GITHUB_ENV
+        shell: bash
+
+      - name: Print Base and Head SHA
         run: |
           echo "Pull Request Base SHA: ${{ github.event.pull_request.base.sha }}"
           echo "Pull Request Head SHA: ${{ github.event.pull_request.head.sha }}"
-          git diff --name-only --diff-filter=AM --find-renames --find-copies ${{ github.event.pull_request.base.sha }}..${{ github.event.pull_request.head.sha }} | grep 'games/.*\.js$' | xargs
-          js_files=$(git diff --name-only --diff-filter=AM --find-renames --find-copies ${{ github.event.pull_request.base.sha }}..${{ github.event.pull_request.head.sha }} | grep 'games/.*\.js$' | xargs)
-          echo "FILES=$js_files" >> $GITHUB_ENV
+
+      - name: Inject MOSS User ID into moss.pl
+        run: |
+          sed -i "s/\$userid=;/\$userid=547529030;/" .github/scripts/moss.pl
```

Comment:
don't we have to hide the MOSS User ID from the public?

Reply:
Yes, we should ideally hide the MOSS User ID, but due to GitHub's limitation on using secrets in workflows from forked repositories, we cannot. Secrets are not accessible in this case, so we have to expose the ID temporarily for the workflow to function.

More details on this limitation can be found here: [GitHub: Using Secrets in Workflows](https://docs.github.com/en/actions/security-for-github-actions/security-guides/using-secrets-in-github-actions#using-secrets-in-a-workflow).